### PR TITLE
Rename `exists` property to `isPresent`

### PR DESCRIPTION
**Pull request by magistrula**
### Purpose
- Rename `exists` to `isPresent` (follow-up on #323)
- Add `exists` to `dsl.js` so that is `isPresent()` can be called on pages and their subcomponents

### Test Coverage
- Updated existing tests
- Added new tests for dsl changes

FYI @YoranBrondsema

### CODE REVIEW

File: addon/-private/dsl.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-import { isHidden } from './properties/is-hidden';
-import { contains } from './properties/contains';
-import { clickOnText } from './properties/click-on-text';
+import { as } from './properties/as';
 import { clickable } from './properties/clickable';
+import { clickOnText } from './properties/click-on-text';
+import { contains } from './properties/contains';
 import { fillable } from './properties/fillable';
+import { isHidden } from './properties/is-hidden';
+import { isPresent } from './properties/is-present';
```
Comment (magistrula) [inline addon/-private/dsl.js]: Added `isPresent` and alphabetized the rest.

File: addon/-private/properties/is-present.js [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
+ * Validates if any element matching the target selector is rendered in the DOM.
+ *
+ * `isPresent` vs. `isVisible`:
+ *   - Both validate that an element matching the target selector can be found in the DOM
+ *   - `isVisible` additionally validates that all matching elements are visible
+ *
+ * Some uses cases for `isPresent` over `isVisible`:
+ *   - To check for the presence of a tag that is never visible in the DOM (e.g., <meta>).
+ *   - To validate that, even though an element may not currently be visible, it is still in the DOM.
+ *   - To validate that an element has not merely been hidden but has in fact been removed from the DOM.
```
Comment (magistrula) [inline addon/-private/properties/is-present.js]: I really liked the idea of calling out the difference between `isVisible` and `isPresent` and wanted to add a bit more detail to the explanation.

File: tests/acceptance/default-properties-test.js [hunk 3]
```diff
@@ -49,33 +50,37 @@ test('Adds default properties', function(assert) {
...
-      },
       contains() {
         return 'contains';
       },
+      isHidden() {
+        return 'isHidden';
+      },
+      isPresent() {
+        return 'isPresent';
+      },
```
Comment (magistrula) [inline tests/acceptance/default-properties-test.js]: Added `isPresent` and alphabetized the rest.

File: tests/integration/default-properties-test.js [hunk 4]
```diff
@@ -38,31 +39,35 @@ test('Overrides default properties', function(assert) {
...
-      },
       contains() {
         return 'contains';
       },
+      isHidden() {
+        return 'isHidden';
+      },
+      isPresent() {
+        return 'isPresent';
+      },
```
Comment (magistrula) [inline tests/integration/default-properties-test.js]: Added `isPresent` and alphabetized the rest.

File: tests/unit/-private/properties/dsl-test.js [hunk 5]
```diff
@@ -28,7 +28,20 @@ moduleForProperty('dsl', function(test) {
...
+      }
+    });
+
+    this.adapter.createTemplate(this, page, 'Lorem <span>ipsum</span>');
+
+    assert.ok(page.isPresent, 'page is rendered in DOM');
+    assert.ok(page.foo.isPresent, 'component is rendered in DOM');
+  });
+
+  ['click', 'clickOn', 'contains', 'fillIn', 'isHidden', 'isPresent', 'isVisible', 'select', 'text', 'value'].forEach((prop) => {
```
Comment (magistrula) [inline tests/unit/-private/properties/dsl-test.js]: Added `isPresent` and alphabetized the rest.

File: tests/unit/addon-exports-test.js [hunk 6]
```diff
@@ -40,7 +40,8 @@ const HELPER_METHODS = [
 const EXTEND_METHODS = HELPER_METHODS.concat('registerExecutionContext');
 
 const EXPECTED_MACROS = [
-  'alias'
+  'alias',
+  'getter'
```
Comment (magistrula) [inline tests/unit/addon-exports-test.js]: I should have included this in #317. Ok to sneak it in here?

File: addon/index.js [hunk 7]
```diff
@@ -10,8 +10,8 @@ import { fillable }    from './-private/properties/fillable';      export { fill
 import { hasClass }    from './-private/properties/has-class';     export { hasClass };
 import { is }          from './-private/properties/is';            export { is };
 import { isHidden }    from './-private/properties/is-hidden';     export { isHidden };
+import { isPresent }      from './-private/properties/is-present';        export { isPresent };
```
Comment (YoranBrondsema) [inline addon/index.js]: Small thing but could you fix the spacing to make it consistent with the other lines?

Comment (magistrula) [reply addon/index.js]: Oh yeah, absolutely...thx.

File: addon/-private/properties/is-present.js (hunk 2 above)
Comment (san650) [reply addon/-private/properties/is-present.js]: Totally agreed!

File: tests/unit/addon-exports-test.js (hunk 6 above)
Comment (san650) [reply tests/unit/addon-exports-test.js]: No problem to include this in this PR :)

**Pull request merged**